### PR TITLE
TransformerEngine FP8 quant backend

### PR DESCRIPTION
**Pull request by bghira**
This pull request introduces expanded support for new quantization presets, especially FP8 and dynamic quantization modes, improves configuration handling for SDNQ and TorchAO quantization, and adds new dependency options. The changes enhance flexibility and compatibility with modern accelerators (e.g., NVIDIA Hopper, Ada Lovelace), improve argument normalization, and provide better integration with distributed training.

**Quantization Preset Expansion and Configuration:**

* Added new quantization presets for FP8 and dynamic quantization, including `"fp8-sdnq"`, `"fp8-native"`, `"fp8-torchao"`, `"fp8wo-torchao"`, `"fp8-int4-torchao"`, `"fp8-transformerengine"`, `"int8dq-torchao"`, and `"int8dq-int4-torchao"`, with corresponding updates in preset lists and configuration logic. (F5219ea8L62R62, [[1]](diffhunk://#diff-4b7d60ab636776e7b0907d9874d2e2f8cb0c6f576ec7e01a7b46a33270301ef1L41-R50) [[2]](diffhunk://#diff-b1e7c0be21b43a43948fe6f04ff01641e8890b137f8cd8c3d2300fe420411198R23-R65) [[3]](diffhunk://#diff-b1e7c0be21b43a43948fe6f04ff01641e8890b137f8cd8c3d2300fe420411198L205-R235)
* Updated TorchAO quantization config mapping and logic to support new dynamic activation and FP8 modes, including additional arguments for versioning and weight dtype. [[1]](diffhunk://#diff-b1e7c0be21b43a43948fe6f04ff01641e8890b137f8cd8c3d2300fe420411198R215-R218) [[2]](diffhunk://#diff-b1e7c0be21b43a43948fe6f04ff01641e8890b137f8cd8c3d2300fe420411198L205-R235)
* Added a new SDNQ preset for native FP8 matmul, targeting H100/H200-class accelerators.

**Dependency and Environment Handling:**

* Introduced a new optional dependency for `transformer_engine` and added it to the `setup.py` extras. [[1]](diffhunk://#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R79) [[2]](diffhunk://#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R372)
* Updated environment variable mappings to include new SDNQ options for more granular configuration via environment files.
* Set `TORCH_BLAS_PREFER_CUBLASLT` to prefer cublasLt for improved performance by default.

**SDNQ and Argument Normalization Improvements:**

* Improved normalization and parsing of SDNQ-related command-line arguments, supporting flexible input types for module lists and config dictionaries.

**TorchAO and DDP Integration Enhancements:**

* Added functions to mark TorchAO parameters for DDP ignore and to log storage summaries, improving distributed training compatibility and debuggability.
* Updated TorchAO filter logic to only quantize `Linear` modules and to check for correct dimensionality.

**Model and Training Logic Updates:**

* Improved LoRA target layer selection for `ltxvideo2` models, filtering out audio-related layers when data has no audio.
* Set `DDP_FIND_UNUSED_PARAMETERS` default to `False` in `ModelFoundation`, which can improve DDP training efficiency.
* Removed outdated Dynamo backend safety check for unsupported quantization modes.
* Updated TorchAO import and usage logic for new quantization classes. [[1]](diffhunk://#diff-b1e7c0be21b43a43948fe6f04ff01641e8890b137f8cd8c3d2300fe420411198R3-R11) [[2]](diffhunk://#diff-b1e7c0be21b43a43948fe6f04ff01641e8890b137f8cd8c3d2300fe420411198L523-R610)

### CODE REVIEW

**Comment by bghira** [comment]
@copilot resolve the merge conflicts in this pull request

**Comment by mhirki** [comment]
Copilot seems to have nuked over 1000 lines from OPTIONS.md while resolving the merge conflict.

**Comment by bghira** [comment]
@mhirki thanks, fixed. i knew it would fail but i'd gone to bed as i submitted the request and realised i should have based on a different commit id instead.

anyway, this is now ready for further testing. the Linears are replaced by TE's module, wrapping the model forward() in the mandatory transformerengine autocast context

additionally, pytorch math backend is adjusted for nvidia devices to use cuBLASLt by default instead of cuBLAS, which gives us a fair speed-up on fp8 gemm.

**Comment by mhirki** [comment]
I tested fp8-sdnq with Ideogram4 but it was very slow with compile enabled. I added these lines to my config:
```
    "--base_model_precision": "fp8-sdnq",
    "--sdnq_compile_mode": "compile",
```

The speed was between 40s/it and 60s/it with gradient accumulation 4. Previously the average speed was around 15s/it. I let it run for 72 steps but it was still slow at that point.

Changing `sdnq_compile_mode` to `eager` improved the speed but it still looked a bit slow.

Lycoris LoKr was working together with fp8-sdnq which is nice because it wasn't working together with Fp8Linear from `simpletuner/helpers/models/ideogram/quantized_loading.py`.

I didn't test TransformerEngine because it refused to install due to not being able to find nccl.h. Eventually, I found a workaround here: https://github.com/NVIDIA/TransformerEngine/issues/2331#issuecomment-3764034876

Now I'm also rebuilding my VAE cache so it's gonna take some time until I can test again.
